bot.py

```diff
@@ -162,6 +168,124 @@ async def get_last_match(ctx, name: str):
+        # initializing total number of players for counting both incoming and existing summoners
+        total_number_of_players = 0
+
+        # initializing server id to a variable
+        server_id = str(ctx.guild.id)
+
+        # storing the json file into a variable
+        if os.path.getsize(json_path) > 0:
+            with open(json_path, "r") as file:
+                file_data = json.load(file)
+
+            # if server id exist in json, add number of players
+            if server_id in file_data:
+                total_number_of_players += len(file_data[server_id])
+
+            # remove summoners from incoming data if it exists in json file with same server ID
+            for player_name in player_list:
+                if any(
+                    player_name in player["user_name"]
+                    for player in file_data[server_id]
+                ):
+                    player_list.remove(player_name)
+
+        # add number of summoners from incoming data to total number of players
+        total_number_of_players += len(player_list)
+
+        if total_number_of_players > MAX_NUM_PLAYERS_TEAM:
+            raise Exception
+
+        # make dictionary for newly coming in players
+        players_list_info = create_summoner_list(player_list, server_id)
+
+        await ctx.send(f"Total Number of Summoners: {total_number_of_players}")
+
+        # if no file exist in path or if the file is empty, dump incoming data
+        if os.path.getsize(json_path) == 0 or server_id not in file_data:
+            with open(json_path, "w") as file:
+                json.dump(players_list_info, file, indent=4)
+                file_data = players_list_info
+
+        # append data to the matching server id
+        else:
+            file_data[server_id] += players_list_info[server_id]
+
+            with open(json_path, "w") as file:
+                json.dump(file_data, file, indent=4)
+
+        embed_data = EmbedData()
+        embed_data.title = "List of Summoners"
+        embed_data.description = "** **"
+        embed_data.color = discord.Color.dark_gray()
+
+        output_str = ""
+
+        for count in range(len(file_data[server_id])):
+
+            output_str += "`{0}{1}` {2}\n".format(
+                file_data[server_id][count]["tier_division"][0],
+                TIER_RANK_MAP.get(file_data[server_id][count]["tier_rank_number"]),
+                file_data[server_id][count]["formatted_user_name"],
+            )
+
+        embed_data.fields = []
+        embed_data.fields.append(
+            {"name": "Summoners", "value": output_str, "inline": False}
+        )
+
+        await ctx.send(embed=create_embed(embed_data))
+    # pylint: disable=broad-except
+    except Exception as e_values:
+        print(str(e_values))
+
+        if "404" in str(e_values):
+            error_title = "Invalid Summoner Name"
+            error_description = "You have entered an invalid summoner name!"
+        elif not str(e_values):
```

Comment:
you are just checking if `e_values` not string here? what if other error comes out that is not related to `limit of 10 summoners`?

Reply:
created an error name and caught error type in exception
e.g.  `elif "Limit Exceeded" in str(e_values):`